CORE-13070 schema_registry/api-doc: update schema docs

pgellert opened the pull request:
All of the changes are fixes to outdated information. All of these have been supported.

* support import mode
* support the force query parameter on mode endpoints
* fix response type for the `GET /schemas/ids/{id}` endpoint
* support specifying id and version on the request when registering a schema

<!--
See https://github.com/redpanda-data/redpanda/blob/dev/CONTRIBUTING.md#pull-request-body
for more details and examples of what is expected in a PR body.

Content in this top section is REQUIRED. Describe, in plain language, the motivation
behind the change (bug fix, feature, improvement) in this PR and how the included
commits address it.

Add the GitHub keyword `Fixes` to link to bug(s) this PR will fix, e.g.
  Fixes #ISSUE-NUMBER, Fixes #ISSUE-NUMBER, ...

If this PR is a backport, link to the original with `Backport of PR`, e.g.
  Backport of PR #PR-NUMBER
-->

Fixes https://redpandadata.atlassian.net/browse/CORE-13070

## Backports Required

<!-- Checking at least one of the checkboxes is REQUIRED if this PR is not a backport. -->

- [ ] none - not a bug fix
- [ ] none - this is a backport
- [x] none - issue does not exist in previous branches
- [ ] none - papercut/not impactful enough to backport
- [ ] v25.2.x
- [ ] v25.1.x
- [ ] v24.3.x

## Release Notes
* none
<!--
If the changes in this PR do not need to be mentioned in the release
notes, then don't add a sub-section and simply list `none`, e.g.

* none

Otherwise, adding a sub-section or `none` is REQUIRED if the PR is not a backport PR.
If this is a backport PR, adding contents to this section will override
the release notes section inherited from the original PR to dev.

Add one or more of the sub-sections with a short description bullet
point of the change, e.g.

### Bug Fixes

* Short description of the bug fix if this is a PR to `dev` branch.

### Features

* Short description of the feature. Explain how to configure.

### Improvements

* Short description of how this PR improves existing behavior.

-->

BenPope commented on src/v/pandaproxy/api/api-doc/schema_registry.json:
```diff
@@ -221,6 +221,13 @@
+          },
+          {
+            "name": "force",
+            "description": "Whether to force update if the preconditions for changing the mode are not met (eg. when setting mode to IMPORT and schemas currently exist)",
```
I'll defer to docs, but:
```suggestion
            "description": "Whether to force update if the preconditions for changing the mode are not met (e.g., when setting mode to IMPORT and schemas currently exist)",
````
Although it might be clearer to state the preconditions, if they're all known, e.g.:
```suggestion
            "description": "Whether to force setting mode to IMPORT when there are existing schemas",
```

BenPope reviewed (commented):
Is it worth bumping the version to `1.1.1`?

A subset should be backported.

vbotbuildovich commented:
<!--TEST RESULTS-->
<!--7ee12c49-5de1-4096-87ef-b0b654d82594-->
<!--TEST RESULTS-->
<!--9035d77f-3cc9-435b-92ed-21c4f04b76e9-->
<!--title-->
## CI test results
<!--/title-->
<!--body-->
<details>
<summary>test results on <a href="https://buildkite.com/redpanda/redpanda/builds/70628">build#70628</a></summary>


| test_class                     | test_method         | test_arguments            | test_kind   | job_url                                                                                   | test_status   | passed   | reason                                                                                                                                                                 |
|:-------------------------------|:--------------------|:--------------------------|:------------|:------------------------------------------------------------------------------------------|:--------------|:---------|:-----------------------------------------------------------------------------------------------------------------------------------------------------------------------|
| CloudStorageScrubberTest       | test_scrubber       | {"cloud_storage_type": 2} | integration | https://buildkite.com/redpanda/redpanda/builds/70628#01989d9b-0ead-4dd0-a6a7-450b363ea3b3 | FLAKY         | 20/21    | upstream reliability is '98.67724867724867'. current run reliability is '95.23809523809523'. drift is 3.43915 and the allowed drift is set to 50. The test should PASS |
| DataMigrationsMultiClusterTest | test_basic          | null                      | integration | https://buildkite.com/redpanda/redpanda/builds/70628#01989d98-9f43-4944-a58a-6f92481bf7df | FLAKY         | 20/21    | upstream reliability is '97.11538461538461'. current run reliability is '95.23809523809523'. drift is 1.87729 and the allowed drift is set to 50. The test should PASS |
| PartitionBalancerTest          | test_rack_awareness | null                      | integration | https://buildkite.com/redpanda/redpanda/builds/70628#01989d98-9f44-4b78-b443-7959050b0911 | FLAKY         | 20/21    | upstream reliability is '100.0'. current run reliability is '95.23809523809523'. drift is 4.7619 and the allowed drift is set to 50. The test should PASS              |
</details>
<details>
<summary>test results on <a href="https://buildkite.com/redpanda/redpanda/builds/70646">build#70646</a></summary>


| test_class            | test_method                          | test_arguments   | test_kind   | job_url                                                                                   | test_status   | passed   | reason                                                                                                                                                                 |
|:----------------------|:-------------------------------------|:-----------------|:------------|:------------------------------------------------------------------------------------------|:--------------|:---------|:-----------------------------------------------------------------------------------------------------------------------------------------------------------------------|
| DataMigrationsApiTest | test_creating_and_listing_migrations | null             | integration | https://buildkite.com/redpanda/redpanda/builds/70646#01989ec4-0624-408b-a3b3-3d5dc61908d0 | FLAKY         | 20/21    | upstream reliability is '99.74226804123711'. current run reliability is '95.23809523809523'. drift is 4.50417 and the allowed drift is set to 50. The test should PASS |
</details>
<!--/body-->

pgellert committed:
schema_registry/api-doc: update schema docs

All of the changes are fixes to outdated information. All of these have
been supported.

* fix response type for the `GET /schemas/ids/{id}` endpoint
* support specifying id and version on the request when registering a
  schema
* list possible values of compatibility levels

pgellert committed:
schema_registry/api-doc: support import mode

pgellert commented:
[force-push](https://github.com/redpanda-data/redpanda/compare/1656a829512d181157bb4901f94622622ee074e3..b1de7513a5699e17a88d06c2d05d71efc747b942): address feedback + also list compatibility level possible values

pgellert commented:
> A subset should be backported.

@BenPope I've split the commit into two now. I'll manually backport the first commit: https://github.com/redpanda-data/redpanda/pull/27216/commits/0db0aeea2911b05a8110c3d05785f045bea87c18.

kbatuigas commented on src/v/pandaproxy/api/api-doc/schema_registry.json:
```diff
@@ -221,6 +221,13 @@
+          },
+          {
+            "name": "force",
+            "description": "Whether to force setting mode to IMPORT when there are existing schemas",
```
```suggestion
            "description": "Whether to force setting the mode to IMPORT even when a schema already exists.",
```
Let me know if this changes the meaning too much. I also had a question regarding "override" in the other PR

pgellert replied:
I went with "If true, override the emptiness check when setting the global mode to IMPORT" after all. Let me know if that makes sense

kbatuigas commented on src/v/pandaproxy/api/api-doc/schema_registry_definitions.def.json:
```diff
@@ -79,15 +79,37 @@
         "compatibilityLevel": {
-          "type": "string"
+          "type": "string",
+          "description": "Compatibility Level",
```
```suggestion
          "description": "Compatibility level",
```
nit; we usually try to default to sentence casing in the docs if it's not a page title

pgellert commented:
[force-push](https://github.com/redpanda-data/redpanda/compare/b1de7513a5699e17a88d06c2d05d71efc747b942..f4728e19e06bb24550b7896166971e1b4231e884): address docs feedback
